airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java

```diff
@@ -674,9 +675,27 @@ public Info info() {
     return new KubePodProcessInfo(podDefinition.getMetadata().getName());
   }
 
+  private Container getMainContainerFromPodDefinition() {
+    final Optional<Container> containerOptional = podDefinition.getSpec()
+        .getContainers()
+        .stream()
+        .filter(c -> c.getName().contentEquals(MAIN_CONTAINER_NAME))
+        .findFirst();
+    if (containerOptional.isEmpty()) {
+      throw new RuntimeException(String.format("Could not find main container definition for pod: %s", podDefinition.getMetadata().getName()));
```

Comment:
Nit: Can we turn this into a WARN message? Although main is something we always have, I'm not sure yet this is a constraint we want to enforce.

Reply:
done